src/shared/axis/axislabel.cpp

```
@@ -0,0 +1,99 @@
+    std::transform(ticks.begin(), ticks.end(), std::back_inserter(ticksWithIndex), [index = 0](const AxisTick& tick) mutable {
+        return TickAndIndex { tick, index++ };
+    });
+    removeOverlappingTicks(ticksWithIndex, labelSize, axisSize);
+
+    std::vector<std::string> labels;
+    labels.resize(ticks.size());
+    int numDecimalDigits = 0;
+    while (numDecimalDigits <= maxDecimalDigits) {
+        labels.clear();
+        for (const auto& tick : ticksWithIndex) {
+            labels[tick.index] = valueToLabel(tick.tick.val, numDecimalDigits);
```

Cppcheck (2.20.0)

[error] Out of bounds access in expression 'labels[tick.index]' because 'labels' is empty.

(containerOutOfBounds)